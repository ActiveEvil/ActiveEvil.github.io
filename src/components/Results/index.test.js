import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import Component from './';

describe('Results component', () => {
  test('does not render when there are no character matches', () => {
    const wrapper = shallow(
      <Component
        characters={[]}
      />
    );

    expect(wrapper.find('.container').exists()).toBe(false);
  });

  test('renders a results list when there are character matches', () => {
    const wrapper = shallow(
      <Component
        characters={[{
          name: 'IG-88',
        }, {
          name: 'R2-D2',
        }]}
      />
    );

    expect(wrapper.find('.list').exists()).toBe(true);
    expect(wrapper.find('li').length).toEqual(2);
  });

  test('renders as expected', () => {
    const component = renderer.create(
      <Component
        characters={[{
          name: 'IG-88',
        }, {
          name: 'R2-D2',
        }]}
      />
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
