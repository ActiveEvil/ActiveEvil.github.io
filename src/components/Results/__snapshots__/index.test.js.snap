// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component renders as expected 1`] = `
<ul
  className="list"
>
  <li
    className="item"
  >
    <a
      className="link"
      onClick={[Function]}
    >
      IG-88
    </a>
  </li>
  <li
    className="item"
  >
    <a
      className="link"
      onClick={[Function]}
    >
      R2-D2
    </a>
  </li>
</ul>
`;
